components/supervisor/frontend/src/index.ts

```diff
@@ -10,9 +10,46 @@
  */
 import { IDEMetricsServiceClient, MetricsName } from "./ide/ide-metrics-service-client";
 IDEMetricsServiceClient.addCounter(MetricsName.SupervisorFrontendClientTotal).catch(() => {})
-window.addEventListener('error', () => {
-    IDEMetricsServiceClient.addCounter(MetricsName.SupervisorFrontendErrorTotal).catch(() => {})
-})
+
+declare global {
+    interface Window {
+        gitpodIDEErrors: Map<string, string>[];
+    }
+}
+
+window.addEventListener('error', (event) => {
+
+    const flags = new Map();
+
+    //@ts-ignore We take a look at what is the resource that was attempted to load;
+    const resourceSource = event?.target?.getAttribute('src') || event?.target?.getAttribute('href');
+
+    // If the event has a `target`, it means that it wasn't a script error
+    if (resourceSource) {
+        let resource = 'unknown';
+        if (resourceSource.match(new RegExp(/\/build\/ide\/code:.{1,100}\/__files__\//g))) {
```

Comment:
Do we really need to limit to 100? .+ won’t work? Why 100?

Reply:
100 seemed just like a number that could cover this whole part, since it can vary in length. I acknowledge that it's not wise to get an abitrary number in here, so thanks for teaching me about the `+` operator in regex!

Reply:
Here's the regex (with simple tests): https://regexr.com/6s8pf

Reply:
And here's the fixing commit: [`6184f1b` (#12222)](https://github.com/gitpod-io/gitpod/pull/12222/commits/6184f1b06b3f9636af9e0ef8c2b6e66a193198f9).